iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/VoltageLevelImpl.java

```diff
@@ -618,4 +619,91 @@ public void allocateVariantArrayElement(int[] indexes, int sourceIndex) {
+                    ConfiguredBusImpl configuredBus = new ConfiguredBusImpl(bus.getId(), bus.getOptionalName().orElse(null), bus.isFictitious(), this);
+                    newTopologyModel.addBus(configuredBus);
+                }
+
+                // transfer retained switches
+                for (Switch sw : topologyModel.getBusBreakerView().getSwitchStream().toList()) {
+                    String busId1 = topologyModel.getBusBreakerView().getBus1(sw.getId()).getId();
+                    String busId2 = topologyModel.getBusBreakerView().getBus2(sw.getId()).getId();
+                    // no notification, this is just a transfer
+                    ((NodeBreakerTopologyModel) topologyModel).removeSwitchFromTopology(sw.getId(), false);
+                    newTopologyModel.addSwitchToTopology((SwitchImpl) sw, busId1, busId2);
+                }
+
+                // reconnect all connectable to new topology model
+                // first store all bus/breaker topological infos associated to this terminal because we will start moving
+                // terminal from old mode to new one, it will modify the old topology model
+                List<Triple<TerminalExt, Bus, Boolean>> oldTopologyModelInfos = new ArrayList<>();
+                for (Terminal oldTerminal : topologyModel.getTerminals()) {
+                    Bus connectableBus = oldTerminal.getBusBreakerView().getConnectableBus();
+                    boolean connected = oldTerminal.isConnected();
+                    oldTopologyModelInfos.add(Triple.of((TerminalExt) oldTerminal, connectableBus, connected));
+                }
+
+                for (var triple : oldTopologyModelInfos) {
+                    TerminalExt oldTerminalExt = triple.getLeft();
+                    Bus connectableBus = triple.getMiddle();
+                    boolean connected = triple.getRight();
+
+                    // if there is no way to find a connectable bus, remove the connectable
+                    // an alternative would be to connect them all to a new trash configured bus
+                    if (connectableBus != null) {
+                        AbstractConnectable<?> connectable = oldTerminalExt.getConnectable();
+
+                        // create the new terminal with new type
+                        TerminalExt newTerminalExt = null;
+                        if (oldTerminalExt.getConnectable().getType() != IdentifiableType.BUSBAR_SECTION) {
+                            newTerminalExt = new TerminalBuilder(networkRef, this, oldTerminalExt.getSide())
+                                    .setBus(connected ? connectableBus.getId() : null)
+                                    .setConnectableBus(connectableBus.getId())
+                                    .build();
+                        }
+
+                        // first attach new terminal to connectable and to new topology model
+                        if (newTerminalExt != null) {
```

Comment:
if the BBS is dealt with in the other if, `newTerminalExt` cannot be null

Reply:
Yes